teku/src/main/java/tech/pegasys/teku/cli/subcommand/MigrateDatabaseCommand.java

```diff
@@ -0,0 +1,186 @@
+
+  // batch size param
+  @CommandLine.Option(
+      names = {"--batch-size"},
+      paramLabel = "<integer>",
+      hidden = true,
+      description = "MB per transaction (Default: 50)",
+      arity = "1")
+  private Integer batchSize = 100;
+
+  @Override
+  public void run() {
+    // validate output format
+    final Optional<DatabaseVersion> maybeOutputVersion = DatabaseVersion.fromString(toDbVersion);
+    if (maybeOutputVersion.isEmpty()) {
+      SUB_COMMAND_LOG.exit(2, "Invalid database version specified: " + toDbVersion);
+    }
+    // validate there is no old database instance present
+    if (Files.isDirectory(dataOptions.getDataBasePath().resolve("beacon.old"))) {
+      SUB_COMMAND_LOG.exit(
+          1,
+          "There is an existing folder in "
+              + dataOptions.getDataBasePath().resolve("beacon.old").toFile()
+              + ", review this folder and remove before continuing.");
+    }
+    // validate source database exists
+    final DatabaseVersion sourceDatabaseVersion =
+        confirmAndCheckOriginalDb(maybeOutputVersion.get());
+
+    final DatabaseMigrater dbMigrater =
+        DatabaseMigrater.builder()
+            .dataOptions(dataOptions)
+            .dataStorageOptions(dataStorageOptions)
+            .eth2NetworkOptions(eth2NetworkOptions)
+            .batchSize(batchSize)
+            .statusUpdater(SUB_COMMAND_LOG::display)
+            .build();
+
+    try {
+      dbMigrater.migrateDatabase(sourceDatabaseVersion, maybeOutputVersion.get());
+      SUB_COMMAND_LOG.display("SUCCESS.");
+      SUB_COMMAND_LOG.display(
+          "The original database is stored in: " + dbMigrater.getMovedOldBeaconFolderPath());
+      SUB_COMMAND_LOG.display("This can be removed once the new database has been verified.");
+    } catch (DatabaseMigraterError error) {
+      SUB_COMMAND_LOG.error("FAILED to migrate database: " + error.getMessage());
+      SUB_COMMAND_LOG.display(
+          "There is a partially created database at: " + dbMigrater.getNewBeaconFolderPath());
+      SUB_COMMAND_LOG.display("This is not in use and could be cleaned up.");
+      System.exit(1);
+    }
+  }
+
+  private DatabaseVersion confirmAndCheckOriginalDb(final DatabaseVersion databaseVersion) {
+    // validate source database exists
+    final DatabaseVersion sourceDatabaseVersion = validateSourceDatabase(databaseVersion);
+
+    displaySourceDatabaseDetails(sourceDatabaseVersion);
+    SUB_COMMAND_LOG.display("Requested database version: " + databaseVersion);
+    SUB_COMMAND_LOG.display("A beacon.new folder will be created with the new database");
+    SUB_COMMAND_LOG.display("If the data is moved successfully: ");
+    SUB_COMMAND_LOG.display(" - The existing beacon folder will become beacon.old");
+    SUB_COMMAND_LOG.display(" - The beacon.new folder will become beacon (the active database)");
+    SUB_COMMAND_LOG.display(
+        " - Once the new database is tested successfully, you can remove the old database");
```

Comment:
This sounds like we'll automatically test the database.  Probably needs to be more like:
```suggestion
        " - Once you have confirmed the new database works, you can remove the old database");
```